dev/release/utils-prepare.sh

```diff
@@ -16,32 +16,53 @@
 # under the License.
 
 ADBC_DIR="${SOURCE_DIR}/../.."
+source "${SOURCE_DIR}/versions.env"
 
 update_versions() {
-  local base_version=$1
-  local next_version=$2
-  local type=$3
+  local type=$1
 
+  local conda_version="${VERSION_NATIVE}"
+  local csharp_version="${VERSION_CSHARP}"
+  local rust_version="${VERSION_CSHARP}"
   case ${type} in
     release)
-      local version=${base_version}
-      local conda_version=${base_version}
-      local docs_version=${base_version}
-      local py_version=${base_version}
-      local r_version=${base_version}
+      local cmake_version="${VERSION_NATIVE}"
+      local docs_version="${RELEASE}"
+      local glib_version="${VERSION_NATIVE}"
+      local java_version="${VERSION_JAVA}"
+      local linux_version="${VERSION_NATIVE}"
+      local py_version="${VERSION_NATIVE}"
+      local r_version="${VERSION_R}"
       ;;
     snapshot)
-      local version=${next_version}-SNAPSHOT
-      local conda_version=${next_version}
-      local docs_version="${next_version} (dev)"
-      local py_version="${next_version}dev"
-      local r_version="${base_version}.9000"
+      local cmake_version="${VERSION_NATIVE}-SNAPSHOT"
+      local docs_version="${RELEASE} (dev)"
+      local glib_version="${VERSION_NATIVE}-SNAPSHOT"
+      local java_version="${VERSION_JAVA}-SNAPSHOT"
+      local linux_version="${VERSION_NATIVE}-SNAPSHOT"
+      local py_version="${VERSION_NATIVE}dev"
+      local r_version="${VERSION_R}.9000"
+      ;;
+    *)
+      echo "Unknown type: ${type}"
+      exit 1
       ;;
   esac
-  local major_version=${version%%.*}
+
+  header "Updating versions for release ${RELEASE}"
+  echo "CMake: ${cmake_version}"
```

Comment:
How about `C` not `CMake` because we use `c/` as directory?

```suggestion
  echo "C: ${c_version}"
```

Reply:
(I'll fix up the PR later tonight!)